src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs

```diff
@@ -50,6 +52,30 @@ public static int RunProcess (string name, string args, DataReceivedEventHandler
 			}
 		}
 
+		static string GetOSBinDirName ()
+		{
+			if (Path.DirectorySeparatorChar == '\\')
+				return "";
+			string os = null;
+			DataReceivedEventHandler output = (o, e) => {
+				if (string.IsNullOrWhiteSpace (e.Data))
+					return;
+				os = e.Data.Trim ();
+			};
+			DataReceivedEventHandler error = (o, e) => {};
+			int r = RunProcess ("uname", "-s", output, error);
```

Comment:
How is this going to work on windows? Is the expectation that it will just return ""? Based on the `Path.DirectorySeparatorChar` check? 
Not sure I like the `Path.DirectorySeparatorChar` I would rather have a proper call to check the Environment Platform instead as its more obvious what is going on.